src/CommonLib/hostfxr_utility.cpp

```diff
@@ -429,105 +429,130 @@ HOSTFXR_UTILITY::FindDotnetExePath(
+            if (index == -1)
+            {
+                break;
+            }
+            if (FAILED(hr = struDotnetSubstring.Copy(struDotnetLocationsString.QueryStr(), index - prevIndex)))
+            {
+                goto Finished;
+            }
+            prevIndex = index;
 
-    // Check the bitness of the currently running process
-    // matches the dotnet.exe found. 
-    if (!IsWow64Process(GetCurrentProcess(), &fIsWow64Process))
-    {
-        // Calling IsWow64Process failed
-        hr = HRESULT_FROM_WIN32(GetLastError());
-        goto Finished;
-    }
-    if (fIsWow64Process)
-    {
-        // 32 bit mode
-        fIsCurrentProcess64Bit = FALSE;
-    }
-    else
-    {
-        SYSTEM_INFO systemInfo;
-        GetNativeSystemInfo(&systemInfo);
-        fIsCurrentProcess64Bit = systemInfo.wProcessorArchitecture == PROCESSOR_ARCHITECTURE_AMD64;
+            if (GetBinaryTypeW(struDotnetSubstring.QueryStr(), &dwBinaryType) &&
+                fIsCurrentProcess64Bit == (dwBinaryType == SCS_64BIT_BINARY)) {
+                // Found a valid dotnet.
+                struDotnetPath->Copy(struDotnetSubstring);
+                fFound = TRUE;
+            }
+        }
     }
 
+Fallback:
+
+    // Look in ProgramFiles
     while (!fFound)
     {
-        index = struDotnetLocationsString.IndexOf(L"\r\n", prevIndex);
-        if (index == -1)
-        {
-            break;
-        }
-        if (FAILED(hr = struDotnetSubstring.Copy(struDotnetLocationsString.QueryStr(), index - prevIndex)))
+        if (FAILED(struDotnetSubstring.Resize(dwPathSize)))
         {
             goto Finished;
         }
-        prevIndex = index;
 
-        if (!GetBinaryTypeW(struDotnetSubstring.QueryStr(), &dwBinaryType))
+        // Program files will changes based on the 
+        dwNumBytesRead = GetEnvironmentVariable(L"ProgramFiles", struDotnetSubstring.QueryStr(), dwPathSize);
+        if (dwNumBytesRead == 0)
         {
-            // TODO should we ignore this failure and continue to trying the next dotnet.exe?
             hr = HRESULT_FROM_WIN32(GetLastError());
             goto Finished;
         }
-        if (fIsCurrentProcess64Bit == (dwBinaryType == SCS_64BIT_BINARY)) {
-            // Found a valid dotnet.
+        else if (dwNumBytesRead == dwPathSize)
+        {
+            dwPathSize *= 2 + 30; // for dotnet substring
+        }
+        else
+        {
+            if (FAILED(hr = struDotnetSubstring.SyncWithBuffer()) ||
+                FAILED(hr = struDotnetSubstring.Append(L"\\dotnet\\dotnet.exe")))
+            {
+                goto Finished;
+            }
+            if (!UTILITY::CheckIfFileExists(struDotnetSubstring.QueryStr()))
+            {
+                hr = HRESULT_FROM_WIN32( GetLastError() );
+                goto Finished;
+            }
             struDotnetPath->Copy(struDotnetSubstring);
             fFound = TRUE;
         }
     }
 
-    if (!fFound)
```

Comment:
Do we have window event log for not being able to find dotnet.exe

Reply:
I believe it is part of this PR pending a bit of refactoring: https://github.com/aspnet/IISIntegration/pull/533/files#diff-093466e281c521516f40989e2d5cb1caR302